packages/components/src/components/post-megadropdown/post-megadropdown.tsx

```diff
@@ -1,56 +1,113 @@
-import { Component, Host, Method, h } from '@stencil/core';
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Method,
+  State,
+  forceUpdate,
+} from '@stencil/core';
 
 @Component({
   tag: 'post-megadropdown',
+  styleUrl: 'post-megadropdown.scss',
   shadow: false,
-  styleUrl: './post-megadropdown.scss',
 })
 export class PostMegadropdown {
   private popoverRef: HTMLPostPopovercontainerElement;
+  private animationClass: string | null = null;
+
+  @Element() host: HTMLPostMegadropdownElement;
+
+  /**
+   * Holds the current visibility state of the dropdown.
+   * This state is internally managed to track whether the dropdown is open (`true`) or closed (`false`),
+   * and updates automatically when the dropdown is toggled.
+   */
+  @State() isVisible: boolean = false;
+
+  /**
+   * Emits when the dropdown is shown or hidden.
+   * The event payload is a boolean: `true` when the dropdown was opened, `false` when it was closed.
+   **/
+  @Event() toggleMegadropdown: EventEmitter<boolean>;
+
+  componentDidLoad() {
+    this.popoverRef.addEventListener('postToggle', (event: CustomEvent<boolean>) => {
+      this.isVisible = event.detail;
+      this.toggleMegadropdown.emit(this.isVisible);
+    });
+  }
 
   /**
-   * Show megadropdown
-   * @param element HTMLElement
-   * @returns boolean
+   * Toggles the dropdown visibility based on its current state.
    */
   @Method()
-  async show(element: HTMLElement) {
-    return this.popoverRef.show(element);
+  async toggle(target: HTMLElement) {
+    this.isVisible ? await this.hide() : await this.show(target);
   }
 
   /**
-   * Hide megadropdown
-   * @returns boolean
+   * Displays the popover dropdown
+   *
+   * @param target - The HTML element relative to which the popover dropdown should be displayed.
    */
   @Method()
-  async hide() {
-    return this.popoverRef.hide();
+  async show(target: HTMLElement) {
+    if (this.popoverRef) {
+      await this.popoverRef.show(target);
+      this.animationClass = 'slide-in';
+    } else {
+      console.error('show: popoverRef is null or undefined');
+    }
   }
 
   /**
-   * Toggle megadropdown
-   * @param element HTMLElement
-   * @param force boolean
-   * @returns boolean
+   * Hides the popover dropdown
    */
   @Method()
-  async toggle(element: HTMLElement, force?: boolean) {
-    return this.popoverRef.toggle(element, force ?? undefined);
+  async hide() {
+    if (this.popoverRef) {
+      await this.popoverRef.hide();
+    } else {
+      console.error('hide: popoverRef is null or undefined');
+    }
   }
 
   private handleBackButtonClick() {
-    this.hide();
+    this.animationClass = 'slide-out';
+    forceUpdate(this);
+    setTimeout(() => {
+      this.hide();
+    }, 350);
```

Comment:
350 ms is the time it takes to make the slide out animation.

Reply:
Then we should add an animation end event listener. And if we change the animation duration in the future, we will not run in a problem, because we forgot to update this timeout.
https://developer.mozilla.org/en-US/docs/Web/API/Element/animationend_event